### ISSUE TITLE
Cannot redeclare function doesn't take into account if function_exists check

### ISSUE THREAD

**Issue by Misplon** [sheabunge/code-snippets#100]
Hi, thanks for your plugin, I've recommended it to users for years.

Hopefully, I haven't missed a setting or something of that nature, if I have, please, let me know.

At some point, a change was made that prevents functions from being used even if they're wrapped in a `function_exists` check.

For example, if in a theme you have:

```
if ( ! function_exists( 'my_theme_function' ) ) :
/**
 * My theme function.
 */
function my_theme_function() {
```

The same function can be safely used in a child theme. This used to be the case for Code Snippets. Not being able to quickly and easily reuse theme functions removes a great deal of functionality from Code Snippets.

I guess this change was made in 2.14.0: `* Added: Basic error checking for duplicate functions and classes.`

For interest sake, [here](https://wordpress.org/support/topic/cannot-redeclare-pluggable-function-of-parent-theme/) is the same request on the .org forum.

**Comment by sheabunge** [comment]
Thanks for reporting this @Misplon. Sorry it's taken so long to address!

I've made an attempt to fix this in 5cb762a0abbc9979a54c3ae7a732268e8c849a42. It's not a perfect solution, as it just ignores any functions that exists in `function_exists` checks, but I figure a simpler solution is less likely to cause issues.

**Comment by Misplon** [comment]
Hi Shea

Thanks for taking a look at this, it's most appreciated.

Cheers :)
